src/Microsoft.Data.Analysis/DataFrame.IO.cs

```diff
@@ -183,140 +183,141 @@ private static DataFrameColumn CreateColumn(Type kind, string[] columnNames, int
                 throw new ArgumentException(string.Format(Strings.ExpectedEitherGuessRowsOrDataTypes, nameof(guessRows), nameof(dataTypes)));
             }
 
-            var linesForGuessType = new List<string[]>();
-            long rowline = 0;
-            int numberOfColumns = dataTypes?.Length ?? 0;
-
-            if (header == true && numberOfRowsToRead != -1)
+            List<DataFrameColumn> columns;
+            string[] fields;
+            using (var textReader = wrappedReader.GetTextReader())
             {
-                numberOfRowsToRead++;
-            }
+                TextFieldParser parser = new TextFieldParser(textReader);
+                parser.SetDelimiters(separator.ToString());
 
-            List<DataFrameColumn> columns;
-            // First pass: schema and number of rows.
-            string line = null;
+                var linesForGuessType = new List<string[]>();
+                long rowline = 0;
+                int numberOfColumns = dataTypes?.Length ?? 0;
 
-            var enumerator = lines.GetEnumerator();
-            while (enumerator.MoveNext())
-            {
-                line = enumerator.Current;
-                if ((numberOfRowsToRead == -1) || rowline < numberOfRowsToRead)
+                if (header == true && numberOfRowsToRead != -1)
+                {
+                    numberOfRowsToRead++;
+                }
+
+                // First pass: schema and number of rows.
+                while ((fields = parser.ReadFields()) != null)
                 {
-                    if (linesForGuessType.Count < guessRows || (header && rowline == 0))
+                    if ((numberOfRowsToRead == -1) || rowline < numberOfRowsToRead)
                     {
-                        var spl = line.Split(separator);
-                        if (header && rowline == 0)
+                        if (linesForGuessType.Count < guessRows || (header && rowline == 0))
                         {
-                            if (columnNames == null)
+                            string[] spl = fields;
```

Comment:
Is this extra variable necessary? Why not just use `fields` below?

Repeated in the block below as well, that reads all rows.